src/main/java/com/google/sps/data/SavedRecipe.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+import com.google.auto.value.AutoValue;
+/**
+ * Abstract class representing saved recipe
+ */
+@AutoValue
+public abstract class SavedRecipe {
+  public abstract long getRecipeId();
+  public abstract String getTitle();
+  public abstract String getImage();
+  public abstract String getSource();
+  public abstract boolean getVegetarian();
+  public abstract boolean getVegan();
+  public abstract boolean getGlutenFree();
+  public abstract boolean getDairyFree();
```

Comment:
I wonder if it's worth to use a set of enum to store all these properties.

Reply:
I created an enum called Diet and I think it probably is more worth is to use a set of enum because it makes the java much cleaner. It might make adding more dietary options in the future easier as well. I changed the other code in the javascript/java files related to profile to reflect this change to use enum as well.